pkg/apis/pravega/v1beta1/pravegacluster_types.go

```diff
@@ -1194,6 +1206,71 @@ func (p *PravegaCluster) validateConfigMap() error {
 	return nil
 }
 
+// ValidateSegmentStoreMemorySettings checks whether the user has passed the required values for segment store memory settings.
+// The required values includes: SegmentStoreResources.limits.memory, pravegaservice.cache.size.max , -XX:MaxDirectMemorySize and -Xmx.
+// Once the required values are set, the method also checks whether the following conditions are met:
+// Total Memory > JVM Heap (-Xmx) + JVM Direct Memory (-XX:MaxDirectMemorySize)
+// JVM Direct memory > Segment Store read cache size (pravegaservice.cache.size.max).
+func (p *PravegaCluster) ValidateSegmentStoreMemorySettings() error {
+	if p.Spec.Pravega.SegmentStoreResources == nil {
+		return fmt.Errorf("Missing required value for field spec.pravega.segmentStoreResources.limits.memory")
+	}
+
+	totalMemoryLimitsQuantity := p.Spec.Pravega.SegmentStoreResources.Limits[corev1.ResourceMemory]
```

Comment:
We also need a check that `p.Spec.Pravega.SegmentStoreResources.Limits != nil` and that `p.Spec.Pravega.SegmentStoreResources.Requests != nil` before we go ahead and check for memory settings (the motivation being that the user might set requests without limits or limits without requests).

Reply:
I agree that `limits` should not be `null`. We could leave `requests` nullable, and if so, set `requests=limits` as it seems to be the underlying behavior in Kubernetes itself.

Reply:
There still needs to be a check for a `nil` `Requests` object before we try to assign `Requests[corev1.ResourceMemory]` to anything (like we currently do it on the next line)

Reply:
As discussed , I will be adding a `nil` check for both `Requests` and `Limits` Objects.